elastic/kibana
# Catching saved object not found error, do not delete referenced package assets

## Summary

Fixes https://github.com/elastic/kibana/issues/142112

Handle Fleet errors more robustly.

### 1. Catching package assets not found error and logging a warning, to prevent it blocking setup/install.
### 2. In assets cleanup logic, doing a check that the assets are not referenced by packages before deleting them.

Verifying 1. and 2. hard, because I don't have a way to reproduce the assets not found error. Tried to install/delete different versions of a package, but didn't run into the issues.

### 3. Force removing a package deletes package policies as well

To verify:
- add at least one integration (e.g. system package)
- force delete integration with API
```
DELETE http://elastic:changeme@localhost:5601/julia/api/fleet/epm/packages/system/1.20.4
kbn-xsrf: kibana

{ "force": true }
```
- verify that package policies are deleted as well
- there is an info log added:
```
[2022-11-24T14:10:29.138+01:00][INFO ][plugins.fleet] deleting package policies of system package because force flag was enabled: 84a4e5d4-e363-4045-9240-9ab151f2b376,6d34a544-7467-48fd-a779-c2d30d808aa6
```

### 4. Catching saved object not found error when tagging package assets
To verify:
- follow the steps above to force delete a package
- add system integration again 
- verify that it succeeds, and there is a warning log
```
[2022-11-24T14:07:47.960+01:00][WARN ][plugins.fleet] Error: Saved object [dashboard/system-01c54730-fee6-11e9-8405-516218e3d268] not found
```

### Checklist
